### ISSUE TITLE
Can't resolve: (#)

### ISSUE THREAD

**Issue by cgonyeo** [idubinskiy/schematyper#8]
```
derek@proton 2 ~/go/src/github.com/coreos/ignition/schema> schematyper ignition.json      
2017/02/03 11:25:45 Can't resolve: (#)
```

I get this error when I try to give schematyper my schema. Any clue how to fix this?

This is `ignition.json`, if it helps: http://lpaste.net/5721431062916104192

**Comment by cgonyeo** [comment]
Oh also fwiw https://github.com/xeipuuv/gojsonschema ingests this json file without issue

**Comment by idubinskiy** [comment]
@dgonyeo So, there were two issues here.

1. This utility expects all properties and definitions to be inside the `"properties"` or `"definitions"` objects. Having `ignition`, `storage`, `filesystems`, etc. at the top level was causing issues. A fix for this is to move those into a top-level `"definitions"` object, like so: http://lpaste.net/6280590999756472320

Making that change led me to discover:

2. `directories/items` was a reference to `directory`, which was a reference to `node`. The utility was not able to handle transitive references. This is now fixed in 2cf8c7fcad09a805cd5cce2ea5361a525d60d198.
